src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/DeprecationCapability.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using NuGet.Versioning;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class DeprecationCapability : IDeprecated
+    {
+        private static readonly string StarAll = VersionRangeFormatter.Instance.Format("p", VersionRange.Parse("*"), VersionRangeFormatter.Instance);
+        private static readonly string StarAllFloating = VersionRangeFormatter.Instance.Format("p", VersionRange.Parse("*-*"), VersionRangeFormatter.Instance);
+
+        public DeprecationCapability(PackageDeprecationMetadataContextInfo deprecatedInfo)
+        {
+            _deprecated = deprecatedInfo;
+        }
+
+        private PackageDeprecationMetadataContextInfo _deprecated;
```

Comment:
We can remove this variable and use an auto property since this instance variable is never used other than for the getter/setter assignment.